src/ape/api/providers.py

```diff
@@ -953,7 +953,16 @@ def send_transaction(self, txn: TransactionAPI) -> ReceiptAPI:
         )
 
         receipt = self.get_receipt(txn_hash.hex(), required_confirmations=required_confirmations)
-        receipt.raise_for_status()
+        if receipt.failed:
+            txn_dict = receipt.transaction.dict()
+            txn_params = cast(TxParams, txn_dict)
+
+            # Replay txn to get revert reason
+            try:
+                self.web3.eth.call(txn_params)
+            except Exception as err:
+                raise self.get_virtual_machine_error(err) from err
+
```

Comment:
Making a bit of a note for myself that I'll need to update [this](https://github.com/ApeWorX/ape/pull/1125/files#diff-4935e80bc2017bbba87c3d13c1c8278742732ffce86ed8286ec6dcec6082f6e7L168) on my side - I think I can just take the exception returned by `get_virtual_machine_error()` and stick the `receipt` on it

I may have to add `receipt` to the constructors for both `VirtualMachineError` and `ContractLogicError` in that case